### ISSUE TITLE
valueUpdater is not a function error using .With and Optional<bool> and null coalescing Operator

### ISSUE THREAD

**Issue by calcock** [ProductiveRage/Bridge.Immutable#5]
I think this one might be a bridge issue, but not sure.

When using the `.With()` Method to set an `Optional<bool>` property if the value comes from a null coalescing operator then the With call fails with an error 'valueUpdater is not a function' (or similar - it seems to depend on the browser)
Reproduction case:

```
 [Ready]
        public static void Main()
        {
            var t = new TestImmutable(true);
            bool? nullableBool = DateTime.Now.Second % 2 == 0;

            var opt1 = nullableBool.HasValue ? nullableBool.Value : Optional<bool>.Missing; // this one works
            var opt2 = (Optional<bool>)(nullableBool ?? Optional<bool>.Missing); // this one does
            var opt3 = nullableBool ?? Optional<bool>.Missing; // this one doesn't

            var t1 = t.With(_ => _.Value, opt1);
            Console.WriteLine("T1: " + t1.Value);

            var t2 = t.With(_ => _.Value, opt2);
            Console.WriteLine("T2: " + t2.Value);

            var t3 = t.With(_ => _.Value, opt3); //breaks: valueUpdater is not a function / Function Expected (or similar)
            Console.WriteLine("T3: " + t3.Value);
        }

        private class TestImmutable : IAmImmutable
        {
            public TestImmutable(Optional<bool> value)
            {
                this.CtorSet(_ => _.Value, value);
            }
            public Optional<bool> Value { get; private set; }
        }
```

In all cases the IDE correctly interprets the type of opt1, opt2 and opt3 as being Optional<bool>, however if the value if from a null coalescing operator it won't work without an explicit cast.

**Comment by calcock** [comment]
Changing from use of `var` to `Optional<bool>` fixes it - this leads me to think that it is probably a Bridge issue rather than an Immutable issue

**Comment by calcock** [comment]
Changing from use of `var` to `Optional<bool>` fixes it - this leads me to think that it is probably a Bridge issue rather than an Immutable issue

**Comment by calcock** [comment]
Raised with Bridge team as https://github.com/bridgedotnet/Bridge/issues/2011

**Comment by LeonidVeriga** [comment]
bridgedotnet/Bridge#2011 fixed and will be included into **Bridge 15.4**

**Comment by ProductiveRage** [comment]
Excellent news! :) I will close this.
